.circleci/config.yml

```diff
@@ -160,524 +220,261 @@ jobs:
       - run:
           name: install python
           command: |
             pyenv install << parameters.python-version>> -s
-
       - save_cache:
           paths:
             - ~/.pyenv
-          key: pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.2.0
-
-      - run: &osx-install-template
+          key: pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.5.1
+      - run:
           name: install
           command: |
+            eval "$(pyenv init --path)"
             eval "$(pyenv init -)"
             pyenv local << parameters.python-version >>
             python -m venv env
             . env/bin/activate
+            pip install pip --upgrade
             pip install -r requirements.txt
             pip install -r tests/requirements.txt
-
-      - run: &unix-build-ext-template
-          name: build
-          command: |
-            . env/bin/activate
-            python setup.py build_ext --inplace
-
+            pip install dimod --no-index -f dist/ --force-reinstall --no-deps
       - run:
           name: run tests
           command: |
             . env/bin/activate
-            python --version
-            coverage run -m unittest discover
-
+            cd tests/
+            coverage run --source=dimod --append -m unittest
       - run:
           name: codecov
           command: |
             . env/bin/activate
+            cd tests/
             codecov
 
-  deploy-osx:
-    parameters:
-      python-version:
-        type: string
-
-    macos:
-      xcode: "12.2.0"
-
-    environment:
-      HOMEBREW_NO_AUTO_UPDATE: 1
-
-      # Force (lie about) macOS 10.9 binary compatibility.
-      # Needed for properly versioned wheels.
-      # See: https://github.com/MacPython/wiki/wiki/Spinning-wheels
-      MACOSX_DEPLOYMENT_TARGET: 10.9
+  test-sdist:
+    docker:
+      - image: circleci/python:3.9
 
     steps:
       - checkout
-
-      - run:
-          name: install pyenv
-          command: |
-            brew install pyenv
-
-      - restore_cache:
-          keys:
-            - pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.2.0
-
-      - run:
-          name: install python
-          command: |
-            pyenv install << parameters.python-version>> -s
-
-      - save_cache:
-          paths:
-            - ~/.pyenv
-          key: pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.2.0
-
-      - run: *osx-install-template
-
-      - run:
-          name: create bdist_wheel
-          command: |
-            . env/bin/activate
-            python -m pip install wheel
-            python setup.py bdist_wheel
-
+      - attach_workspace:
+          at: dist
       - run:
-          name: deploy
+          name: install from sdist
           command: |
+            python -m venv env
             . env/bin/activate
-            python -m pip install twine
-            twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing ./dist/*
+            pip install dist/dimod*.tar.gz
```

Comment:
Don't you want to run unittests after install as well?

Reply:
This is just checking that it's installable. Though I suppose we can run unittests as well.